test_vm/tests/commit_post_test.rs

```diff
@@ -0,0 +1,426 @@
+use fil_actor_cron::Method as CronMethod;
+use fil_actor_market::Method as MarketMethod;
+use fil_actor_miner::{
+    aggregate_pre_commit_network_fee, max_prove_commit_duration,
+    new_deadline_info_from_offset_and_epoch, power_for_sector, DeadlineInfo, Method as MinerMethod,
+    PoStPartition, PowerPair, PreCommitSectorBatchParams, ProveCommitSectorParams,
+    SectorPreCommitInfo, SectorPreCommitOnChainInfo, State as MinerState, SubmitWindowedPoStParams,
+};
+use fil_actor_power::{
+    CreateMinerParams, CreateMinerReturn, Method as PowerMethod, State as PowerState,
+    UpdateClaimedPowerParams,
+};
+use fil_actor_reward::Method as RewardMethod;
+use fil_actors_runtime::cbor::serialize;
+use fil_actors_runtime::runtime::Policy;
+use fil_actors_runtime::{test_utils::*, BURNT_FUNDS_ACTOR_ADDR};
+use fil_actors_runtime::{
+    CRON_ACTOR_ADDR, REWARD_ACTOR_ADDR, STORAGE_MARKET_ACTOR_ADDR, STORAGE_POWER_ACTOR_ADDR,
+    SYSTEM_ACTOR_ADDR,
+};
+use fvm_ipld_bitfield::BitField;
+use fvm_ipld_blockstore::MemoryBlockstore;
+use fvm_ipld_encoding::{BytesDe, RawBytes};
+use fvm_shared::address::Address;
+use fvm_shared::bigint::Zero;
+use fvm_shared::clock::ChainEpoch;
+use fvm_shared::econ::TokenAmount;
+use fvm_shared::error::ExitCode;
+use fvm_shared::randomness::Randomness;
+use fvm_shared::sector::{
+    PoStProof, RegisteredPoStProof, RegisteredSealProof, SectorNumber, StoragePower,
+};
+use fvm_shared::METHOD_SEND;
+use num_traits::sign::Signed;
+use test_vm::util::{apply_ok, create_accounts};
+use test_vm::{ExpectInvocation, TEST_VM_RAND_STRING, VM};
+
+#[test]
+fn commit_post_flow_happy_path() {
+    let store = MemoryBlockstore::new();
+    let mut v = VM::new_with_singletons(&store);
+    let addrs = create_accounts(&v, 2, TokenAmount::from(10_000e18 as i128));
+    let seal_proof = RegisteredSealProof::StackedDRG32GiBV1P1;
+    let (owner, worker) = (addrs[0], addrs[0]);
+    let (id_addr, robust_addr) = create_miner(
+        &mut v,
+        owner,
+        worker,
+        seal_proof.registered_window_post_proof().unwrap(),
+        TokenAmount::from(10_000e18 as i128),
+    );
+    let mut v = v.with_epoch(200);
+
+    // precommit and advance to prove commit time
+    let sector_number: SectorNumber = 100;
+    precommit_sectors(&mut v, 1, 1, worker, id_addr, seal_proof, sector_number, true, None);
+
+    let balances = v.get_miner_balance(id_addr);
+    assert!(!balances.pre_commit_deposit.is_negative());
+
+    let prove_time =
+        v.get_epoch() + max_prove_commit_duration(&Policy::default(), seal_proof).unwrap();
+    let (v, _) = advance_by_deadline_to_epoch(v, id_addr, prove_time);
+
+    // prove commit, cron, advance to post time
```

Comment:
Worth making this its own method in the interest of future tests?

Reply:
Yeah that's going to be the post commit test PR